internal/winapi/utils.go

```diff
@@ -80,3 +83,21 @@ func ConvertStringSetToSlice(buf []byte) ([]string, error) {
 	}
 	return nil, errors.New("string set malformed: missing null terminator at end of buffer")
 }
+
+// parseUtf16LE parses a UTF-16LE byte array into a string (without passing
+// through a uint16 or rune array).
+func ParseUtf16LE(b []byte) string {
+	var sb strings.Builder
+	sb.Grow(len(b) / 2)
+	for len(b) > 0 {
+		r := rune(binary.LittleEndian.Uint16(b))
+		if utf16.IsSurrogate(r) && len(b) > 2 {
+			sb.WriteRune(utf16.DecodeRune(r, rune(binary.LittleEndian.Uint16(b[2:]))))
+			b = b[4:]
+		} else {
+			sb.WriteRune(r)
+			b = b[2:]
```

Comment:
Are we guaranteed that the len of b is at least 2 here?

Reply:
This was old code, I have replaced it with library version now.